zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/UserTaskAssignmentDeniedApplier.java

```diff
@@ -30,6 +30,11 @@ public UserTaskAssignmentDeniedApplier(final MutableProcessingState processingSt
 
   @Override
   public void applyState(final long key, final UserTaskRecord value) {
+    final String deniedReason = value.getDeniedReason();
+    final UserTaskRecord userTask = userTaskState.getUserTask(key);
+    userTask.setDeniedReason(deniedReason);
+    userTaskState.update(userTask);
```

Comment:
❓ Why do we need to store the denied reason in state? Are we planning to use this later again? My expectation is that we respond it to the client and that it's exported on the log for the `UsertTaskRecord:X_DENIED` event.